### PR TITLE
Allow to consider MDs as BIOS RAIDs

### PR DESCRIPTION
**Pull request by joseivanlopez**
PBI: https://trello.com/c/z4eVEC6E/385-https-bugzillasusecom-showbugcgiid1092417

### CODE REVIEW

File: package/yast2-storage-ng.changes [hunk 1]
```diff
@@ -1,3 +1,10 @@
+-------------------------------------------------------------------
+Thu May 17 13:32:37 UTC 2018 - jlopez@suse.com
+
+- Allow to consider MD RAIDs as BIOS RAIDs by using the env
+  variable LIBSTORAGE_MDPART (bsc#1092417).
```
Comment (jreidinger) [inline package/yast2-storage-ng.changes]: please also document variable somewhere. IDeally in suse-doc and in release notes.

File: src/lib/y2storage/md_container.rb [hunk 2]
```diff
@@ -64,8 +64,17 @@ def software_defined?
...
 
+    # Objects of this class are not considered RAIDs at all
     def types_for_is
-      super << :md_container
+      types = super
+
+      types.delete(:disk_device)
+      types.delete(:raid)
+      types.delete(:bios_raid)
+      types.delete(:software_raid)
```
Comment (jreidinger) [inline src/lib/y2storage/md_container.rb]: are you sure about this change? It looks a bit agresive change here.

Comment (joseivanlopez) [reply src/lib/y2storage/md_container.rb]: Yes, it ensures the same behavior than before.

File: src/lib/y2storage/storage_env.rb [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+module Y2Storage
+  # Class to read ENV values for storage
+  class StorageEnv
+    include Singleton
+    include Yast::Logger
+
+    ENV_MULTIPATH = "LIBSTORAGE_MULTIPATH_AUTOSTART".freeze
+
+    ENV_BIOS_RAID = "LIBSTORAGE_MDPART".freeze
```
Comment (jreidinger) [inline src/lib/y2storage/storage_env.rb]: are you sure that linuxrc always export it upper cased? in ruby-bindings we did this: https://github.com/yast/yast-ruby-bindings/blob/master/src/ruby/yast/debugger.rb#L105

Comment (jreidinger) [reply src/lib/y2storage/storage_env.rb]: ah, see it below, so already done.

File: src/lib/y2storage/storage_env.rb [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      # Similar to what linuxrc does, also consider the flag activated if the
+      # variable is used with no value or with "1"
+      value.casecmp?("on") || value.empty? || value == "1"
+    end
+
+    # Read an ENV variable
+    #
+    # @param variable [String]
+    # @return [String, nil]
+    def read(variable)
```
Comment (jreidinger) [inline src/lib/y2storage/storage_env.rb]: looks private

File: src/lib/y2storage/storage_env.rb (hunk 3 above)
Comment (joseivanlopez) [reply src/lib/y2storage/storage_env.rb]: Here we do the same, see `#read`.

File: package/yast2-storage-ng.changes (hunk 1 above)
Comment (joseivanlopez) [reply package/yast2-storage-ng.changes]: This value is not new, we are bringing back it. Is not already documented?

Comment (jreidinger) [reply package/yast2-storage-ng.changes]: Sorry, zero results found with google neither at suse-doc. So we should definitively document it. 
Either installation guide or storage guide.

**Comment by wfeldt** [comment]
Looks basically ok to me but what I really miss are some explanatory comments in the code. Not sure if I remember any of the logic in 2 months...

**Comment by wfeldt** [comment]
lgtm, just please fix the double negation

**Pull request merged**